src/transformation/transformation.ts

```diff
@@ -3,9 +3,305 @@ import {
     ensureNodeDefinition,
     getNodeDefinition,
     Node,
-    NODE_DEFINITION_SYMBOL, registerNodeDefinition,
+    NODE_DEFINITION_SYMBOL, Origin, Property,
+    registerNodeDefinition,
     registerNodeProperty
 } from "../model/model";
+import {Issue, IssueSeverity} from "../validation";
+import {Position} from "../model/position";
+
+function isClassType(type) : boolean {
+    return typeof type == 'function' &&
+        type.name != undefined &&
+        new RegExp(`[class\\s+${type.name}.*]`).test(type.toString());
+}
+
+function isSuperClass(subClass, superClass) : boolean {
+    return isClassType(superClass) &&
+        // eslint-disable-next-line no-prototype-builtins
+        superClass.prototype.constructor.isPrototypeOf(subClass);
+}
```

Comment:
This part looks suspicious and brittle to me. Do we really need to check whether something is a class? In JS there are numerous ways of doing classes, across compilers and runtimes.